File: 264/Solution.go
func nthUglyNumber(n int) int {
    res := make([]int, n)
    n2, n3, n5 := 0, 0, 0
    res[0] = 1
    for i := 1; i < n; i++{
        res[i] = min(res[n2] * 2, min(res[n3] * 3, res[n5] * 5))
        if res[i] == res[n2] * 2 { n2++ }
        if res[i] == res[n3] * 3 { n3++ }
        if res[i] == res[n5] * 5 { n5++ }
    }
    return res[n-1]
}
 func min(a, b int) int{
    if a < b {
        return a
    }
    return b
 }

File: 264/Solution.js
/**
 * @param {number} n
 * @return {number}
 */
var nthUglyNumber = function(n) {
    let res = [1];
    let n2 = 0, n3 = 0, n5 = 0;

    while(res.length < n){
        res.push(Math.min(res[n2]*2, Math.min(res[n3]*3, res[n5]*5)));
        if(res[res.length-1] === res[n2]*2) n2++;
        if(res[res.length-1] === res[n3]*3) n3++;
        if(res[res.length-1] === res[n5]*5) n5++;
    }
    return res[n-1];
};
